nemo/src/io/formats/sparql.rs

```diff
@@ -0,0 +1,172 @@
+//! Handler for resources of type SPARQL (SPARQL query language for RDF)).
+
+use spargebra::Query;
+use std::sync::Arc;
+
+use nemo_physical::{
+    datavalues::{AnyDataValue, DataValue},
+    resource::ResourceBuilder,
+};
+use oxiri::Iri;
+
+use crate::{
+    io::format_builder::{
+        format_parameter, format_tag, value_type_matches, AnyImportExportBuilder, FormatParameter,
+        Parameters, StandardParameter,
+    },
+    rule_model::{
+        components::{import_export::Direction, term::value_type::ValueType},
+        error::validation_error::ValidationErrorKind,
+    },
+    syntax::import_export::{attribute, file_format},
+};
+
+use super::{ExportHandler, FormatBuilder, ImportHandler};
+
+use crate::io::formats::dsv::{value_format::DsvValueFormats, DsvHandler};
+
+/// A char limit to decide if a query should be send via GET or POST
+const HTTP_GET_CHAR_LIMIT: usize = 2000;
+
+format_tag! {
+    pub(crate) enum SparqlTag(SupportedFormatTag::Sparql) {
+        Sparql => file_format::SPARQL,
+    }
+}
+
+format_parameter! {
+    pub(crate) enum SparqlParameter(StandardParameter) {
+        Format(name = attribute::FORMAT, supported_types = &[ValueType::Constant, ValueType::Tuple]),
+        Base(name = attribute::BASE, supported_types = &[ValueType::Constant]),
+        Endpoint(name = attribute::ENDPOINT, supported_types = &[ValueType::Constant]),
+        Query(name = attribute::QUERY, supported_types = &[ValueType::String]),
+    }
+}
+
+impl FormatParameter<SparqlTag> for SparqlParameter {
+    fn required_for(&self, tag: SparqlTag) -> bool {
+        matches!(tag, SparqlTag::Sparql) && matches!(self, Self::Endpoint | Self::Query)
+    }
+
+    fn is_value_valid(&self, value: AnyDataValue) -> Result<(), ValidationErrorKind> {
+        value_type_matches(self, &value, self.supported_types())?;
+
+        match self {
+            SparqlParameter::BaseParamType(base) => {
+                FormatParameter::<SparqlTag>::is_value_valid(base, value)
+            }
+            SparqlParameter::Base => Ok(()),
+            SparqlParameter::Format => DsvValueFormats::try_from(value).and(Ok(())).map_err(|_| {
+                ValidationErrorKind::ImportExportValueFormat {
+                    file_format: "sparql".to_string(),
+                }
+            }),
+            SparqlParameter::Endpoint => value
+                .to_iri()
+                .ok_or(ValidationErrorKind::ImportExportInvalidIri)
+                .and_then(|iri| {
+                    Iri::parse(iri)
+                        .map(|_| ())
+                        .map_err(|_| ValidationErrorKind::ImportExportInvalidIri)
+                }),
```

Comment:
This shouldn't need to throw the same error twice, clearly, `to_iri()` should actually return a validated IRI.